core/src/main/java/com/digitalpebble/storm/crawler/util/MetadataTransfer.java

```diff
@@ -28,37 +28,68 @@
  * what should be stored back in the persistence layer etc...
  **/
 public class MetadataTransfer {
+	/**
+	 * Parameter name indicating which metadata to transfer to the outlinks.
+	 * Boolean value.
+	 **/
+	public static final String metadataTransferParamName = "metadata.transfer";
 
-    private static String metadataTransferParamName = "metadata.transfer";
+	/**
+	 * Parameter name indicating whether to track the url path or not. Value is
+	 * either a vector or a single valued String.
+	 **/
+	public static final String trackPathParamName = "metadata.track.path";
 
-    private List<String> mdToKeep = new ArrayList<String>();
+	/** Metadata key name for tracking the source URLs **/
+	public static final String urlPathKeyName = "url.path";
```

Comment:
So it took me a while to understand what you were doing even though I just requested this. Maybe call this "anchor.path" instead of "url.path" or anchor.text.path